core/factorization/ic.cpp

```diff
@@ -104,7 +104,7 @@ std::unique_ptr<Composition<ValueType>> Ic<ValueType, IndexType>::generate(
     std::shared_ptr<const matrix_type> ic;
     // Compute LC factorization
     if (parameters_.algorithm == incomplete_algorithm::syncfree ||
-        exec == exec->get_master()) {
+        std::dynamic_pointer_cast<const OmpExecutor>(exec)) {
```

Comment:
@upsj is right. It needs to be
```suggestion
        std::dynamic_pointer_cast<const OmpExecutor>(exec) && !std::dynamic_pointer_cast<const ReferenceExecutor>(exec)) {
```
¯\\_(ツ)_/¯